File: 백준/Silver/1931. 회의실 배정/회의실 배정.java
import java.util.Arrays;
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int N = sc.nextInt();
        int[][] meetings = new int[N][2];
        for (int i = 0; i < N; i++) {
            meetings[i][0] = sc.nextInt();
            meetings[i][1] = sc.nextInt();
        }

        Arrays.sort(meetings, (a, b) -> {
            if (a[1] == b[1]) {
                return Integer.compare(a[0], b[0]);
            }
            return Integer.compare(a[1], b[1]);
        });

        int ans = 0;
        int endTime = 0;
        for (int i = 0; i < N; i++) {
            if (meetings[i][0] >= endTime) {
                ans++;
                endTime = meetings[i][1];
            }
        }
        System.out.println(ans);
    }
}

File: 백준/Silver/1931. 회의실 배정/회의실 배정.py
N = int(input())
I = [list(map(int, input().split())) for _ in range(N)]

# 끝나는 시간 기준으로 정렬 (끝나는 시간이 같으면 시작 시간으로 정렬)
I.sort(key=lambda x: (x[1], x[0]))

ans = 0
end_time = 0

for start_time, finish_time in I:
    if start_time >= end_time:  # 현재 회의 시작 시간이 이전 회의 종료 시간 이후라면
        ans += 1
        end_time = finish_time  # 종료 시간을 갱신

print(ans)
